src/components/OcButton.spec.js

```diff
@@ -0,0 +1,109 @@
+import OcButton from './OcButton.vue';
+import {createLocalVue, shallowMount} from '@vue/test-utils';
+const localVue = createLocalVue()
+
+describe('OcButton', () => {
+  const testSlots = { default: '<p class="text">Test button</p>' }
+  const route = {
+    path: '/some/path',
+    name: 'some-route'
+  }
+  function getWrapperWithProps(props) {
+    return shallowMount(OcButton, { localVue: localVue, propsData: props })
+  }
+  function getWrapperWithTestSlot() {
+    return shallowMount(OcButton, { localVue: localVue, slots: testSlots })
+  }
+
+  it('displays slot html', () => {
+    const wrapper = getWrapperWithTestSlot()
+    const slot = wrapper.find('p')
+    expect(slot).toBeTruthy()
+    expect(slot.attributes('class')).toBe('text')
+    expect(slot.text()).toBe('Test button')
+  })
+
+  it('can stop class propagation', () => {
+    const wrapper = getWrapperWithProps({
+      stopClassPropagation: true,
+      size: 'middle'
+    })
+    expect(wrapper.attributes('class')).toContain('')
+  })
+
+  describe('emit click event', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({})
+    })
+    it('emits click event when click is triggered', async () => {
+      await wrapper.trigger('click')
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeTruthy()
+    })
+    it('does not emit click event when input is triggered', async () => {
```

Comment:
How would you ever trigger an `input` on a button? 🤔

Reply:
i've removed, this is making confusions